### DISCUSSION TITLE
Using meta property in table

### DISCUSSION THREAD

**Discussion by Zishan3165** [KevinVandy/material-react-table#393] [Q&A]
I'm trying to use the meta property to pass some common methods like saveCell when bulk editing. But typescript will not allow me because those functions do not exist in TableMeta type in tanstack table. How can I add it so that I can pass columns from a consumer component which can use methods from table.options.meta.saveCell? Any insight or help would be appreciated! New to using this library. I haven't found any other way to achieve this.
-Thanks

**Comment by thafer30** [comment]
Running into this same issue.

Also related https://github.com/KevinVandy/material-react-table/issues/828

**Comment by davegomez** [comment]
I have the same issue: I can't use the `meta` to pass custom data, functions, or components to my `Cell` components because TypeScript complains about the property not existing in the type.

I need help finding something in the MRT documentation suggesting how to use the meta feature or how to declare the types.

@KevinVandy's answer to this issue is not helpful, and I wonder why the usage of `meta` inside the column definitions is not considered important enough to be included in the documentation.

I want to clarify that I'm talking about the `meta` option of the column definitions, not the global `meta` in the table definition.

**Reply by davegomez** [reply]
@KevinVandy, I wonder if you can give us a hand on this? I haven't been able to build the correct TS type to use the [meta data](https://tanstack.com/table/latest/docs/api/core/column-def#meta) in the columns using the MRT exported types or the TanStack one (but even if I could created using the TanStack ones, it doesn't feel right) and I'm stuck at this point.

**Reply by KevinVandy** [reply]
yeah, I guess it's not really supported right now unless I modify some typescript stuff. I haven't really seen the need for meta in MRT, but might as well add it.
